src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -309,10 +341,30 @@ void CoreWorkerDirectActorTaskSubmitter::PushActorTask(const ClientQueue &queue,
         } else if (status.ok()) {
           task_finisher_->CompletePendingTask(task_id, reply, addr);
         } else {
+          // push task failed due to network error. For example, actor is dead
+          // and no process response for the push task.
+          absl::MutexLock lock(&mu_);
+          auto queue_pair = client_queues_.find(actor_id);
+          RAY_CHECK(queue_pair != client_queues_.end());
+          auto &queue = queue_pair->second;
+
+          bool immediately_mark_object_fail = (queue.state == rpc::ActorTableData::DEAD);
           bool will_retry = task_finisher_->PendingTaskFailed(
-              task_id, rpc::ErrorType::ACTOR_DIED, &status);
+              task_id, rpc::ErrorType::ACTOR_DIED, &status, queue.dead_info,
+              immediately_mark_object_fail);
           if (will_retry) {
             increment_completed_tasks = false;
+          } else if (!immediately_mark_object_fail) {
```

Comment:
Why not query the GCS for the actor state instead of waiting for a notification?

Reply:
To avoid too many queries when actor keeps submitting tasks to a DEAD actor.

Reply:
We should only query once for the first task failure right? The rest can wait for the same response.

Reply:
When this actor sends tasks to multiple DEAD actors, the query number will still be large.